src/keys.rs

```diff
@@ -218,26 +223,17 @@ fn check_structural_validity(
     }
 }
 
-/// We currently use `SpendAuth` as the `IssuanceAuth`.
-type IssuanceAuth = SpendAuth;
-
 /// An issuance key, from which all key material is derived.
 ///
 /// $\mathsf{isk}$ as defined in [ZIP 227][issuancekeycomponents].
 ///
 /// [issuancekeycomponents]: https://qed-it.github.io/zips/zip-0227#issuance-key-derivation
-#[derive(Debug, Copy, Clone)]
-pub struct IssuanceAuthorizingKey([u8; 32]);
+#[derive(Copy, Clone)]
+pub struct IssuanceAuthorizingKey(NonZeroScalar);
 
 impl From<SpendingKey> for IssuanceAuthorizingKey {
     fn from(sk: SpendingKey) -> Self {
-        IssuanceAuthorizingKey(*sk.to_bytes())
-    }
-}
-
-impl ConstantTimeEq for IssuanceAuthorizingKey {
-    fn ct_eq(&self, other: &Self) -> Choice {
-        self.to_bytes().ct_eq(other.to_bytes())
+        IssuanceAuthorizingKey::from_bytes(*sk.to_bytes()).unwrap()
```

Comment:
Done

Reply:
Oh I see, missed that. I'll update that

Reply:
Updated this with some changes, that I will summarize here:

- We don't actually need to allow for the conversion from `SpendingKey` to `IssuanceAuthorizationKey` anymore, since we are deriving it directly from ZIP 32 seeds. 
- So I removed the `From<SpendingKey>` implementation altogether, and I instead took the relevant part of logic and moved it to the only place we were using it, which is in the `from_zip32_seed` function.
- In there, I allowed for the generation of an error if the `IssuanceAuthorizingKey::from_bytes` function fails, as pointed out in the original comment in this thread.